editor/plugins/gizmos/navigation_link_3d_gizmo_plugin.cpp

```diff
@@ -70,59 +70,101 @@ void NavigationLink3DGizmoPlugin::redraw(EditorNode3DGizmo *p_gizmo) {
 
 	p_gizmo->clear();
 
-	// Draw line between the points.
-	Vector<Vector3> lines;
-	lines.append(start_position);
-	lines.append(end_position);
-
-	// Draw start position search radius
-	for (int i = 0; i < 30; i++) {
-		// Create a circle
-		const float ra = Math::deg_to_rad((float)(i * 12));
-		const float rb = Math::deg_to_rad((float)((i + 1) * 12));
-		const Point2 a = Vector2(Math::sin(ra), Math::cos(ra)) * search_radius;
-		const Point2 b = Vector2(Math::sin(rb), Math::cos(rb)) * search_radius;
-
-		// Draw axis-aligned circle
-		switch (up_axis) {
-			case Vector3::AXIS_X:
-				lines.append(start_position + Vector3(0, a.x, a.y));
-				lines.append(start_position + Vector3(0, b.x, b.y));
-				break;
-			case Vector3::AXIS_Y:
-				lines.append(start_position + Vector3(a.x, 0, a.y));
-				lines.append(start_position + Vector3(b.x, 0, b.y));
-				break;
-			case Vector3::AXIS_Z:
-				lines.append(start_position + Vector3(a.x, a.y, 0));
-				lines.append(start_position + Vector3(b.x, b.y, 0));
-				break;
-		}
-	}
+	uint32_t points_in_octant = 4;
+	float inc = (Math_PI / (4 * points_in_octant));
 
-	// Draw end position search radius
-	for (int i = 0; i < 30; i++) {
-		// Create a circle
-		const float ra = Math::deg_to_rad((float)(i * 12));
-		const float rb = Math::deg_to_rad((float)((i + 1) * 12));
-		const Point2 a = Vector2(Math::sin(ra), Math::cos(ra)) * search_radius;
-		const Point2 b = Vector2(Math::sin(rb), Math::cos(rb)) * search_radius;
-
-		// Draw axis-aligned circle
+	Vector<Vector3> lines;
+	// points_in_octant * 8 * 2 per circle * 2 circles. 2 for the start-end. 4 for the arrow, and another 4 if bidirectionnal.
+	lines.resize(points_in_octant * 8 * 2 * 2 + 2 + 4 + (link->is_bidirectional() ? 4 : 0));
+	uint32_t index = 0;
+	Vector3 *lines_ptrw = lines.ptrw();
+	// Draw line between the points.
+	lines_ptrw[index++] = start_position;
+	lines_ptrw[index++] = end_position;
+	float search_radius_squared = search_radius * search_radius;
```

Comment:
This should be `real_t`, as should some other variables you added.

Reply:
I kept float because previous code was using float in the loop, but it could be `real_t`, I will change it